### PR TITLE
Spack v0.18

### LINKED ISSUE #161: Spack v0.18.0 was released - update spack fork/branch and use this version for first spack-stack release
spack v0.18.0 was released, and it contains many new features, improvements and bugfixes:

https://github.com/spack/spack/releases/tag/v0.18.0

We should update our fork/branch of spack and use that version in our first spack-stack release in June 2022.

### PR DESCRIPTION
**Pull request by kgerheiser**
Update Spack to v0.18

https://github.com/spack/spack/releases/tag/v0.18.0

* Modules install location is now in modules.yaml
* Reuse is now the default when concretizing
* Concretize option changed in environment. Trying out `when_possible` to reuse when possible.
* Better concretization error messages
* Improvements for external Python linking

Fix #161 

Testing:

- [x] Orion (`ufs-weather-model-env` with `control_c48` run)
- [x] Gaea (built the `jedi-ewok-env` but errors - unrelated to this PR - starting up `ecflow_ui` - see https://github.com/NOAA-EMC/spack-stack/issues/168)
- [x] Full builds on Github Actions with Linux+macOS and GCC, Intel, Clang.

### CODE REVIEW

**Comment by kgerheiser** [comment]
Builds are running

**Comment by kgerheiser** [comment]
All the builds passed except one which failed for an unrelated reason because of a connection error.

**Comment by climbfuji** [comment]
I suggest we add checkboxes to the PR description with the HPCs we want to test? Better than fixing post-commit? We can split the work ...

**Comment by kgerheiser** [comment]
Ok, sounds good. I added some boxes for a couple machines. Testing the build is easy enough. Do you do other testing like running a model?

**Comment by climbfuji** [comment]
> Ok, sounds good. I added some boxes for a couple machines. Testing the build is easy enough. Do you do other testing like running a model?

I usually run the ufs-weather-model `control_c48` run and the `fv3-bundle` ctests. But we don't need to do all of this on every platform. I'll start with gaea. Will also add a few other platforms that JCSDA needs.

**Comment by kgerheiser** [comment]
The `+pmix` variant in external OpenMPI, as on Orion, was removed.

**Comment by kgerheiser** [comment]
Built and ran ufs-weather-model control_c48 on Orion.

**Comment by climbfuji** [comment]
@kgerheiser I think I am ok with merging this. It built on two largely different HPCs with two very different env specs, and it built on your Mac and in the CI tests (correct?)

**Comment by kgerheiser** [comment]
Yes, [Ubuntu Intel](https://github.com/kgerheiser/spack-stack/actions/runs/2419124692), [Ubuntu GCC](https://github.com/kgerheiser/spack-stack/actions/runs/2419124548), [macOS GCC](https://github.com/kgerheiser/spack-stack/actions/runs/2419124827).

[apple-clang](https://github.com/kgerheiser/spack-stack/runs/6682466300?check_suite_focus=true) failed from a network error during a download.


I compiled on my Mac with apple-clang and Orion with Intel.

**Comment by climbfuji** [comment]
And I did gaea with intel as indicated in the PR description above.

File: .gitmodules [hunk 1]
```diff
@@ -3,4 +3,4 @@
   #url = https://github.com/spack/spack
   #branch = develop
   url = https://github.com/NOAA-EMC/spack
-  branch = jcsda_emc_spack_stack
+  branch = jcsda_emc_spack_stack
```
Comment (climbfuji) [inline .gitmodules]:
```suggestion
  branch = jcsda_emc_spack_stack

```

**Review by climbfuji — CHANGES_REQUESTED** [review]
spack submodule pointer is correct, just restore the newline in `.gitmodules` please and we are good to go!

**Comment by kgerheiser** [comment]
But I'm getting a conflict some reason.

**Comment by climbfuji** [comment]
probably need to pull in the updates from spack-stack develop first.

**Comment by kgerheiser** [comment]
This was good to get done. I have a couple of minor things I want to merge now that this is done.

**Comment by climbfuji** [comment]
Same here. Can you please add the corresponding issues/PR to the project kanban (first spack release) so that we can keep track of it? Also, please provide feedback if you think that something shouldn't go into the release that is already in the kanban. Thanks!

**Comment by kgerheiser** [comment]
Yes

**Pull request merged**